store/sqlstore/thread_store.go

```diff
@@ -497,20 +461,15 @@ func (s *SqlThreadStore) GetThreadForUser(teamId string, threadMembership *model
 	}
 
 	var thread JoinedThread
-	query, threadArgs, err := s.getQueryBuilder().
+	query := s.getQueryBuilder().
 		Select("Threads.*, Posts.*").
 		From("Threads").
-		Column(sq.Alias(sq.Expr(unreadRepliesQuery), "UnreadReplies")).
+		Column(sq.Alias(sq.Expr(unreadRepliesQuery, unreadRepliesArgs...), "UnreadReplies")).
```

Comment:
This is a nice improvement as well.

Reply:
So `sq.Expr` creates a `Sqlizer` embedding its own `sql` and `args`. `sq.Alias` then holds onto the `Sqlizer`, and `Column` creates a nested `Sqlizer`. When it comes time to finally build the query, `Sqlizer`'s `ToSQL` extracts the args from the embedded `Sqlizer`:

https://github.com/lieut-data/squirrel/blob/6ba7e8d947599922316c8b509ba57512659959c6/part.go#L17-L30

The `args` array gets built up as we process all of these embedded `Sqlizer` statements:

https://github.com/lieut-data/squirrel/blob/6ba7e8d947599922316c8b509ba57512659959c6/select.go#L88-L93

Long story short, this works:
```go
package main

import (
	"fmt"
	"testing"

	sq "github.com/Masterminds/squirrel"
)

func TestExpr(t *testing.T) {
}

func ExampleExpr() {
	subQuery, subqueryArgs, _ := sq.StatementBuilder.PlaceholderFormat(sq.Question).
		Select("COUNT(Posts.Id)").
		From("Posts").
		Where(sq.NotEq{"DeleteAt": 0}).
		ToSql()

	query, args, _ := sq.StatementBuilder.PlaceholderFormat(sq.Question).
		Select("PostId").
		Column(sq.Alias(sq.Expr(subQuery, subqueryArgs...), "NumPosts")).
		From("Threads").
		Where(sq.Eq{"PostId": "post_id"}).
		ToSql()

	fmt.Println(query)
	fmt.Println(args)

	// Output: SELECT PostId, (SELECT COUNT(Posts.Id) FROM Posts WHERE DeleteAt <> ?) AS NumPosts FROM Threads WHERE PostId = ?
	// [0 post_id]
}
```

```
> go test -v
=== RUN   TestExpr
--- PASS: TestExpr (0.00s)
=== RUN   ExampleExpr
--- PASS: ExampleExpr (0.00s)
PASS
ok      test    0.093s
```

The need to "render" the subquery feels like something else we could add to our fork of squirrel -- it should suffice to just write something like the following (or even inlining the subquery directly!):
```go
subQuery := sq.StatementBuilder.PlaceholderFormat(sq.Question).
        Select("COUNT(Posts.Id)").
        From("Posts").
        Where(sq.NotEq{"DeleteAt": 0})

query, args, _ := sq.StatementBuilder.PlaceholderFormat(sq.Question).
        Select("PostId").
        Column(sq.Alias(sq.Expr(subQuery), "NumPosts")).
        From("Threads").
        Where(sq.Eq{"PostId": "post_id"}).
        ToSql()

```

Reply:
Another way of saying this, is that it aggregates args just like `.Where(...)` adds args to the final array.

Reply:
>  (or even inlining the subquery directly!)

ooh nice I like this!